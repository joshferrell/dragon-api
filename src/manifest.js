const manifest = {
    server: {
        cache: 'redis'
    },
    connections: [
        {
            port: 8000,
            labels: ['api']
        },
        {
            port: 8001,
            labels: ['admin']
        }
    ],
    registrations: [
        { plugin: 'inert' },
        { plugin: 'vision' },
        { plugin: 'hapi-swagger' },
        { plugin: './auth.plugin' }
    ]
};

export default manifest;
