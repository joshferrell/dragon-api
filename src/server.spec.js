import manifest from './manifest';

describe('server configuration', () => {
    it('should include a server manifest', () => {
        expect(manifest).toMatchSnapshot();
    });
});
