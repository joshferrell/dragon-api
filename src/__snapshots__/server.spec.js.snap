// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server configuration should include a server manifest 1`] = `
Object {
  "connections": Array [
    Object {
      "labels": Array [
        "api",
      ],
      "port": 8000,
    },
    Object {
      "labels": Array [
        "admin",
      ],
      "port": 8001,
    },
  ],
  "registrations": Array [
    Object {
      "plugin": "inert",
    },
    Object {
      "plugin": "vision",
    },
    Object {
      "plugin": "hapi-swagger",
    },
    Object {
      "plugin": "./auth.plugin",
    },
  ],
  "server": Object {
    "cache": "redis",
  },
}
`;
